pkg/state/block_differ.go

```diff
@@ -110,20 +110,34 @@ func (d *blockDiffer) createPrevBlockMinerFeeDiff(prevBlockID proto.BlockID, min
 	return diff, minerAddr, nil
 }
 
-func (d *blockDiffer) createTransactionDiff(tx proto.Transaction, block *proto.BlockHeader, height uint64, vrf []byte, initialisation bool) (txBalanceChanges, error) {
+func (d *blockDiffer) createTransactionDiff(tx proto.Transaction, block *proto.BlockHeader, height uint64, vrf []byte, initialisation, status bool) (txBalanceChanges, error) {
 	blockInfo, err := proto.BlockInfoFromHeader(d.settings.AddressSchemeCharacter, block, height, vrf)
 	if err != nil {
 		return txBalanceChanges{}, err
 	}
 	differInfo := &differInfo{initialisation, blockInfo}
-	txChanges, err := d.handler.createDiffTx(tx, differInfo)
+	var txChanges txBalanceChanges
+	if status {
+		txChanges, err = d.handler.createDiffTx(tx, differInfo)
+	} else {
+		txChanges, err = d.handler.createFeeDiffTx(tx, differInfo)
```

Comment:
Looks like `createFeeDiff` is only needed for Invoke and Exchange, since it is called when `status` is false.